federation/pkg/kubefed/unjoin.go

```diff
@@ -81,6 +81,16 @@ func unjoinFederation(f cmdutil.Factory, cmdOut, cmdErr io.Writer, config util.A
 		return nil
 	}
 
+	//we need to ensure deleting the config map created in the deregistered cluster
+	//this configmap was created when the cluster joined this federation to aid
+	//the kube-dns of that cluser to aid service discovery
+	unjoinClusterFactory := config.HostFactory(unjoinFlags.Name, unjoinFlags.Kubeconfig)
+	err = deleteConfigMap(unjoinClusterFactory)
```

Comment:
Please help me understand this better.
What would we safeguard by first removing the value and then deleting the object compared to removing the object directly.